src/terrain/node_group.cpp

```diff
@@ -1,84 +1,129 @@
+NodeGroup::get_adjacent_map() const
+{
+    return std::map<const NodeGroup*, UnitPath>(adjacent.begin(), adjacent.end());
 }
 
-std::map<NodeGroup *, UnitPath> NodeGroup::get_adjacent_map(){
+std::map<NodeGroup*, UnitPath>
+NodeGroup::get_adjacent_map()
+{
     return adjacent;
 }
 
-std::set<const NodeGroup *> NodeGroup::get_adjacent_clear(int path_type)const{
-    std::set<const NodeGroup *> out;
+std::set<const NodeGroup*>
+NodeGroup::get_adjacent_clear(int path_type) const
+{
+    std::set<const NodeGroup*> out;
 
-    for (const std::pair<NodeGroup *,UnitPath> t : adjacent) {
+    for (const std::pair<NodeGroup*, UnitPath> t : adjacent) {
         if (t.second.compatible(path_type) && t.second.is_open()) {
             out.insert(t.first);
         }
     }
     return out;
 }
 
-std::map<NodeGroup *, UnitPath> NodeGroup::merge_groups(NodeGroup other){
-    center_x = (center_x*tiles.size() + other.center_x*other.tiles.size())/(tiles.size()+other.tiles.size());
-    center_y = (center_y*tiles.size() + other.center_y*other.tiles.size())/(tiles.size()+other.tiles.size());
-    center_z = (center_z*tiles.size() + other.center_z*other.tiles.size())/(tiles.size()+other.tiles.size());
+std::map<NodeGroup*, UnitPath>
+NodeGroup::merge_groups(NodeGroup other)
+{
+    center_x = (center_x * tiles.size() + other.center_x * other.tiles.size())
+               / (tiles.size() + other.tiles.size());
+    center_y = (center_y * tiles.size() + other.center_y * other.tiles.size())
+               / (tiles.size() + other.tiles.size());
+    center_z = (center_z * tiles.size() + other.center_z * other.tiles.size())
+               / (tiles.size() + other.tiles.size());
 
-    for (const Tile* t : other.get_tiles()){
+    for (const Tile* t : other.get_tiles()) {
         tiles.insert(t);
     }
-    for (std::pair<NodeGroup *const, UnitPath> adj: other.get_adjacent_map()){
+    for (std::pair<NodeGroup* const, UnitPath> adj : other.get_adjacent_map()) {
         add_adjacent(adj.first, adj.second);
     }
-    path_type_ = path_type_ & other.path_type_; // restriction of the ways to get between tiles
+    path_type_ =
+        path_type_ & other.path_type_; // restriction of the ways to get between tiles
 
     return other.get_adjacent_map();
 }
 
-float NodeGroup::get_center_x(){ return center_x; }
-float NodeGroup::get_center_y(){ return center_y; }
-float NodeGroup::get_center_z(){ return center_z; }
-std::array<float, 3> NodeGroup::sop() const {
-    return { center_x, center_y, center_z };
+float
+NodeGroup::get_center_x()
+{
+    return center_x;
+}
+
+float
+NodeGroup::get_center_y()
+{
+    return center_y;
 }
 
-const std::set<const Tile*, TilePCompare> NodeGroup::get_tiles() const {
+float
+NodeGroup::get_center_z()
+{
+    return center_z;
+}
+
+std::array<float, 3>
+NodeGroup::sop() const
+{
+    return {center_x, center_y, center_z};
+}
+
+const std::set<const Tile*, TilePCompare>
+NodeGroup::get_tiles() const
+{
```

Comment:
Yes more lines of code this is what we need. (also inline)

Reply:
Inlining will be done later